include/CL/sycl/platform/detail/platform_tail.hpp

```diff
@@ -18,32 +18,35 @@ namespace sycl {
     @{
 */
 
-/** Return a list of all available devices
+/** Returns a vector class containing all SYCL devices associated with
+    this SYCL platform.
 
     Return synchronous errors via SYCL exception classes.
 */
 vector_class<device>
 platform::get_devices(info::device_type device_type) const {
-  // Start with the default device
-  vector_class<device> devices = { {} };
+  /** If \c get_devices is called with the host platform,
+      returns the host_device.
+  */
+  if(is_host()) return { {} };
+
+  vector_class<device> devices;
+  device_type_selector s { device_type };
 
 #ifdef TRISYCL_OPENCL
-  // Then add all the OpenCL devices
-  for (const auto &d : boost::compute::system::devices())
-    devices.emplace_back(d);
+  // Add the desired OpenCL devices
+  for (const auto &d : get_boost_compute().devices()) {
+    // Get the SYCL device from the Boost Compute device
+    cl::sycl::device sycl_dev { d };
+    /* Return the devices with the good criterion according to the selector
+       and the \c cl_platform_id of the device platform and the instance
+    */
+    if(s(sycl_dev) >= 0 && sycl_dev.get_platform().get() == get())
```

Comment:
Why do you need `sycl_dev.get_platform().get() == get()`?
By construction, `get_boost_compute().devices()` gives you only the devices of the platform, isn't it?